llvm/lib/Transforms/Scalar/InductiveRangeCheckElimination.cpp

```diff
@@ -989,17 +999,15 @@ bool InductiveRangeCheckElimination::run(
     return false;
   }
 
-  if (!isProfitableToTransform(*L))
-    return false;
```

Comment:
Currently, we have an early exit from the transformation if we determine it's not profitable. Does it make sense to keep this behavior? Is it a common path where we perform an early exit?

Reply:
That is a good point, thanks. I added this early exit back